src/clientiface.cpp

```diff
@@ -101,6 +101,16 @@ void RemoteClient::GetNextBlocks (
 	m_nothing_to_send_pause_timer -= dtime;
 	m_map_send_completion_timer += dtime;
 
+	if (m_map_send_completion_timer > g_settings->getFloat("server_unload_unused_data_timeout") * 0.8f) {
+		infostream << "Server: Player " << m_name << ", RemoteClient " << peer_id
+				<< ": full map send is taking too long "
+				<< m_map_send_completion_timer
+				<< "s, restarting to avoid visible blocks being unloaded."
```

Comment:
```suggestion
		infostream << "Server: Player " << m_name << ", peer_id=" << peer_id
				<< ": full map send is taking too long ("
				<< m_map_send_completion_timer
				<< "s), restarting to avoid visible blocks being unloaded."
```